src/prefect/cli/deployment.py

```diff
@@ -933,7 +934,20 @@ async def delete(
         $ prefect deployment delete --id dfd3e220-a130-4149-9af6-8d487e02fea6
     """
     async with get_client() as client:
-        if name is None and deployment_id is not None:
+        if _all:
+            deployments = await client.read_deployments()
+            if len(deployments) == 0:
+                exit_with_error("No deployments found.")
```

Comment:
```suggestion
                exit_with_success("No deployments found.")
```

i think this should be `exit_with_success` so that the exit code is idempotent